.github/actions/setup_dependencies/action.yml

```diff
@@ -0,0 +1,31 @@
+runs:
+  using: "composite"
+  steps:
+    - name: Install 3rd party from apt
+      run: sudo apt install unar zlib1g-dev liblzo2-dev lzop lziprecover
+      shell: bash
+
+    - name: Get full Python version
+      id: full-python-version
+      shell: bash
+      run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
+
+    - name: Restore pip cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('poetry.lock') }}
+
+    - name: Pip install
+      run: python -m pip install --upgrade pip poetry
+      shell: bash
+
+    - name: Restore poetry cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pypoetry/virtualenvs
```

Comment:
I don't know what poetry is storing on those folders (`cache`, `artifacts`) so I did not wanted to cache them.
As far as I can see, caching the virtualenv is enough, because when restoring it in a fresh CI run as doing a `poetry install` it says that "No dependencies to install or update". 

So if you can say what are those and why should they be cached I would gladly add them, otherwise I think caching only the venv itself is cleaner.